src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -17,54 +19,123 @@ namespace NuGet.PackageManagement.UI
+                        await UpdateMarkdownAsync(ReadmeViewModel.ReadmeMarkdown);
+                    }
+                }).PostOnFailure(nameof(PackageReadmeControl), nameof(ReadmeViewModel_PropertyChanged));
+            }
+            if (e.PropertyName == nameof(ReadmePreviewViewModel.IsBusy))
+            {
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsBusy)));
+            }
+            if (e.PropertyName == nameof(ReadmePreviewViewModel.IsReadmeReady))
+            {
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsReadmeReady)));
             }
         }
 
-        private async Task UpdateMarkdownAsync()
+        private async Task UpdateMarkdownAsync(string markdown)
         {
             try
             {
-                if (!string.IsNullOrWhiteSpace(ReadmeViewModel.ReadmeMarkdown))
+                UpdateBusy(true);
+                if (_markdownPreview is not null)
                 {
-                    await _markdownPreview.UpdateContentAsync(ReadmeViewModel.ReadmeMarkdown, ScrollHint.None);
+                    await TaskScheduler.Default;
+                    await _markdownPreview.UpdateContentAsync(markdown, ScrollHint.None);
                 }
             }
             catch (Exception ex) when (ex is ArgumentException || ex is InvalidOperationException)
             {
-                ReadmeViewModel.ErrorWithReadme = true;
-                ReadmeViewModel.ReadmeMarkdown = string.Empty;
-                await TelemetryUtility.PostFaultAsync(ex, nameof(ReadmePreviewViewModel));
+                if (_cancellationTokenSource.IsCancellationRequested)
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ReadmeViewModel.ErrorWithReadme = true;
+                    ReadmeViewModel.ReadmeMarkdown = string.Empty;
+                    await TelemetryUtility.PostFaultAsync(ex, nameof(ReadmePreviewViewModel));
+                }
+            }
+            finally
+            {
+                if (!_cancellationTokenSource.IsCancellationRequested)
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    UpdateBusy(false);
+                }
             }
         }
 
+        private void UpdateBusy(bool isBusy)
+        {
+            _isBusy = isBusy;
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsReadmeReady)));
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsBusy)));
+        }
+
         private void UserControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
         {
             if (e.OldValue is ReadmePreviewViewModel oldMetadata)
             {
-                ThreadHelper.JoinableTaskFactory.Run(async () =>
+                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
Another fire and forget that adds JTF overhead for no reason, see: https://github.com/NuGet/NuGet.Client/pull/6190#discussion_r1889485483.